packages/client/src/v2-events/features/events/useEvents/procedures/actions/action.ts

```diff
@@ -67,6 +72,42 @@ function errorToastOnConflict(error: TRPCClientError<AppRouter>) {
   }
 }
 
+function getCleanedDeclarationDiff(
+  eventConfiguration: EventConfig,
+  originalDeclaration?: EventState,
+  declarationDiff?: EventState
+): ActionUpdate | undefined {
+  if (isEmpty(declarationDiff)) {
+    return declarationDiff
+  }
+
+  // If there's no original declaration, just clean the update and return it
+  if (isEmpty(originalDeclaration)) {
+    return omitHiddenPaginatedFields(
+      eventConfiguration.declaration,
+      declarationDiff
+    )
+  }
+
+  // Merge original + updates so we get the final event state
+  // (Needed because omitHiddenPaginatedFields func requires a full snapshot, not partial)
+  const merged = deepMerge(originalDeclaration, declarationDiff)
+
+  // Remove any hidden/paginated fields from the merged declaration
+  // (Ensures we only consider fields relevant to the event configuration)
+  const cleanedDeclaration = omitHiddenPaginatedFields(
+    eventConfiguration.declaration,
+    merged
+  )
+
+  // From the update, keep only fields that are valid in the cleaned declaration
+  // (Prevents applying updates to hidden/invalid fields)
+  const cleanedDeclarationDiff: ActionUpdate = Object.fromEntries(
+    Object.entries(declarationDiff).filter(([key]) => key in cleanedDeclaration)
+  )
+  return cleanedDeclarationDiff
```

Comment:
```suggestion
  return Object.fromEntries(
    Object.entries(declarationDiff).filter(([key]) => key in cleanedDeclaration)
  )
```

Reply:
In future debugging, this style ensures we can debug it without any hassle, hence choose this style.